### PR TITLE
JSON export

### LINKED ISSUE #1312: Get JSON export working with new prototype structure.

### PR DESCRIPTION
**Pull request by steffenagger**
Utilizing `Realm.Collection.toJSON` from `realm@10.0.0-beta.11` to generate JSON export & support circular structures.

This closes #1312

### CODE REVIEW

**Review by kneth — APPROVED** [review]
LGTM

File: src/services/data-exporter/json.ts [hunk 1]
```diff
@@ -21,82 +21,43 @@ import Realm from 'realm';
...
+  try {
+    return JSON.stringify(map, null, INDENTATION_SPACES);
+  } catch (err) {
+    if (
+      err instanceof TypeError &&
+      err.message.startsWith('Converting circular structure to JSON')
+    ) {
+      return JSON.stringify(
+        map,
+        Realm.JsonSerializationReplacer,
```
Comment (kraenhansen) [inline src/services/data-exporter/json.ts:40]: Any reason that we're not just always calling with the `JsonSerializationReplacer`? If yes, I think that would be a good candidate for a comment.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,11 +10,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ### Enhancements
 
-- None
+- JSON export now supports circular structures.
```
Comment (kraenhansen) [inline CHANGELOG.md:13]:
```suggestion
- JSON export now supports circular structures. ([#1315](https://github.com/realm/realm-studio/pull/1315))
```

**Review by kraenhansen — CHANGES_REQUESTED** [review]
Looking good! Just two minor suggestions.

File: src/services/data-exporter/json.ts (hunk 1 above)
Comment (steffenagger) [reply src/services/data-exporter/json.ts:40]: It will add `$refId` to entities that does not have a `primaryKey` for reference. I'll add a comment.

Comment (steffenagger) [reply src/services/data-exporter/json.ts:43]: Added.

**Review by kraenhansen — APPROVED** [review]

**Pull request merged**